### PR TITLE
Fixups/v2

### PR DESCRIPTION
**Pull request by victorjulien**
https://github.com/OISF/suricata/pull/10337 and https://github.com/OISF/suricata/pull/10340, plus an added error message.

changes since #10344:
- actually include #10337

### CODE REVIEW

File: src/detect-http-header.c [hunk 1]
```diff
@@ -663,10 +663,11 @@ static InspectionBuffer *GetHttp1HeaderData(DetectEngineThreadCtx *det_ctx, cons
             size_t size = size1 + size2 + 2;
             if (hdr_td->items[i].len < size) {
                 // Use realloc, as this pointer is not freed until HttpMultiBufHeaderThreadDataFree
-                hdr_td->items[i].buffer = SCRealloc(hdr_td->items[i].buffer, size);
-                if (unlikely(hdr_td->items[i].buffer == NULL)) {
+                void *tmp = SCRealloc(hdr_td->items[i].buffer, size);
+                if (unlikely(tmp == NULL)) {
                     return NULL;
                 }
+                hdr_td->items[i].buffer = tmp;
```
Comment (inashivb) [inline src/detect-http-header.c:672]: Just a question: `tmp` is a `void *`, why don't we explicitly cast it to `u8 *` here as `buffer` is `u8 *`?

Comment (victorjulien) [reply src/detect-http-header.c:672]: man realloc: `void *realloc(void *ptr, size_t size);`, so it doesn't matter where you cast it.

**Review by inashivb — APPROVED** [review]
LGTM

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 18265

**Comment by victorjulien** [comment]
Merged in #10347, thanks!

**Pull request closed without merging**